lib/install/management/store_files.go

```diff
@@ -201,20 +214,24 @@ func (d *Dispatcher) getChildren(ds *object.Datastore, dsPath string) ([]string,
 }
 
 // getSortedChildren returns all children under datastore path in reversed order.
-func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string) ([]string, error) {
-	result, err := d.getChildren(ds, dsPath)
+func (d *Dispatcher) getSortedChildren(ds *object.Datastore, dsPath string, matchPattern ...string) ([]string, error) {
+	result, err := d.getChildren(ds, dsPath, matchPattern...)
 	if err != nil {
 		return nil, err
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(result)))
 	return result, nil
 }
 
-func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
+func (d *Dispatcher) lsSubFolder(ds *object.Datastore, dsPath string, matchPattern ...string) (*types.ArrayOfHostDatastoreBrowserSearchResults, error) {
 	defer trace.End(trace.Begin(dsPath))
 
+	// match every subfolers by default
+	if len(matchPattern) == 0 {
+		matchPattern = append(matchPattern, "*")
+	}
 	spec := types.HostDatastoreBrowserSearchSpec{
-		MatchPattern: []string{"*"},
+		MatchPattern: matchPattern,
```

Comment:
actually I'm not sure. So I just tried to create a vch 'dev' with volumeStore being 'dev/abc', thus 'dev/abc/volumes' is excluded. after I delete it without --force, dev/abc still exists, VIC, kvStore, appliance is gone. what's weird is appliance.iso, bootstrap.iso and tether.debug are also not removed.